### PR TITLE
feat: provide helm charts to install merbridge

### PR DESCRIPTION
**Pull request by Xunzhuo**
Resolve: https://github.com/merbridge/merbridge/issues/57
Doc: https://github.com/Xunzhuo/merbridge/tree/helm-charts/deploy
Signed-off-by: Xunzhuo <mixdeers@gmail.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -22,3 +22,14 @@ format-go:
 lint: lint-c lint-go
 
 format: format-c format-go
+
+helm: helm-linkerd helm-istio
+
+helm-linkerd:
+	helm template --set-string "merbridge.namespace=linkerd,merbridge.mode=linkerd" merbridge helm > deploy/all-in-one-linkerd.yaml
```
Comment (kebe7jun) [inline Makefile:29]: can we use `--namespace` instead of `merbridge.namespace`?

File: deploy/README.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+> Note that: excute these commands in root dir of merbridge
+
+### Install Merbridge on Istio
+
++ set merbridge.mode=istio to switch to Istio mode, default value is `istio`.
++ set namespace where merbridge is going to install by --set merbridge.namespace, default value is `istio-system`.
+
+``` bash
+helm install merbridge helm
```
Comment (kebe7jun) [inline deploy/README.md:17]: I think `helm install -n istio-system merbridge merbridge` is better.

File: helm/templates/daemonset.yaml [hunk 3]
```diff
@@ -0,0 +1,82 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: {{ .Values.merbridge.fullname }}
+  namespace: {{ .Values.merbridge.namespace }}
```
Comment (kebe7jun) [inline helm/templates/daemonset.yaml:5]: `{{ .Release.Namespace }}` may be better.

File: helm/templates/daemonset.yaml [hunk 4]
```diff
@@ -0,0 +1,82 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: {{ .Values.merbridge.fullname }}
```
Comment (kebe7jun) [inline helm/templates/daemonset.yaml:4]: `{{ .Release.Name }}`

File: helm/values.yaml [hunk 5]
```diff
@@ -0,0 +1,35 @@
+# Default values for merbridge.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+# some settings of merbridge when start up
+merbridge:
```
Comment (kebe7jun) [inline helm/values.yaml:6]: This is unnecessary.

File: helm/templates/daemonset.yaml [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+            memory: {{ .Values.deployment.resources.container.limit.memory }}
+        securityContext:
+          privileged: true
+        volumeMounts:
+          - mountPath: /sys/fs
+            name: sys-fs
+          - mountPath: /host/ips
+            name: host-ips
+      initContainers:
+      - image: "{{ .Values.deployment.image.repository }}:{{ .Values.deployment.image.tag }}"
```
Comment (kebe7jun) [inline helm/templates/daemonset.yaml:44]:
```suggestion
      - image: "{{ .Values.image.hub }}/{{ .Values.image.repository }}:{{ .Values.image.tag }}"
```

File: helm/values.yaml [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  fullname: merbridge
+  namespace: istio-system
+  mode: istio
+  ipsFilePath: /host/ips/ips.txt
+
+# some settings of deployment
+deployment:
```
Comment (kebe7jun) [inline helm/values.yaml:13]: also unnecessary I think.

File: helm/templates/daemonset.yaml (hunk 3 above)
Comment (Xunzhuo) [reply helm/templates/daemonset.yaml:5]: question is that if we use {{ .Release.Namespace }}, when we generate templates, how to handle the generated namespace?

Comment (Xunzhuo) [reply helm/templates/daemonset.yaml:5]: Any flags or commands to specific the release name/namespace when executing helm template?

File: helm/templates/daemonset.yaml [hunk 8]
```diff
@@ -0,0 +1,82 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: {{ .Release.Name }}
```
Comment (kebe7jun) [inline helm/templates/daemonset.yaml:4]: if we choose to use `{{ .Values.fullname }}`, all of name should be this.

**Comment by kebe7jun** [comment]
generally LGTM.

File: helm/templates/daemonset.yaml (hunk 8 above)
Comment (Xunzhuo) [reply helm/templates/daemonset.yaml:4]: Update by https://github.com/merbridge/merbridge/pull/65#discussion_r810727669

I think maybe we just use values.fullname ?

Comment (kebe7jun) [reply helm/templates/daemonset.yaml:4]: ok. so we should only use `.Values.fullname` rather than `.Release.Name` .

**Comment by Xunzhuo** [comment]
I will start a PR to add CI check to compare if we make changes in helm charts and if we update the generated yamls.

**Review by kebe7jun — APPROVED** [review]
Thanks very much!

**Pull request merged**